cmd/root.go

```diff
@@ -39,54 +42,59 @@ profiles you.`,
 
 var cfgFile string
 
-func init() {
-	binDir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+func newClient(domain, scope string) *client.Client {
+	// For the CLI client, we rely on the admin APIs to generate a CLI token.
+	// We may want in the future rely on OAuth to handle the permissions with
+	// more granularity.
+	c := newAdminClient()
+	token, err := c.GetToken(&client.TokenOptions{
+		Domain:   domain,
+		Subject:  "CLI",
+		Audience: permissions.CLIAudience,
+		Scope:    []string{scope},
+	})
 	if err != nil {
-		panic(err)
+		log.Errorf("Could not generate access to domain %s", domain)
+		log.Error(err)
+		os.Exit(1)
+	}
+	return &client.Client{
+		Domain:        domain,
+		DisableSecure: config.IsDevRelease(),
+		Authorizer:    &request.BearerAuthorizer{Token: token},
 	}
+}
 
+func newAdminClient() *client.Client {
+	var pass []byte
+	if !config.IsDevRelease() {
+		pass = []byte(os.Getenv("COZY_ADMIN_PASSWORD"))
+		if len(pass) == 0 {
+			var err error
+			fmt.Printf("Password:")
+			pass, err = gopass.GetPasswdMasked()
+			if err != nil {
+				panic(err)
+			}
+		}
+	}
+	return &client.Client{
+		Domain:        config.AdminServerAddr(),
+		DisableSecure: config.IsDevRelease(),
+		Authorizer:    &request.BasicAuthorizer{Password: string(pass)},
+	}
+}
+
+func init() {
 	flags := RootCmd.PersistentFlags()
-	flags.StringVarP(&cfgFile, "config", "c", "", "configuration file (default \"$HOME/.cozy/cozy.yaml\")")
+	flags.StringVarP(&cfgFile, "config", "c", "", "configuration file (default \"$HOME/.cozy.yaml\")")
 
 	flags.String("host", "localhost", "server host")
 	checkNoErr(viper.BindPFlag("host", flags.Lookup("host")))
 
 	flags.IntP("port", "p", 8080, "server port")
 	checkNoErr(viper.BindPFlag("port", flags.Lookup("port")))
 
-	flags.String("subdomains", "nested", "how to structure the subdomains for apps (can be nested or flat)")
-	checkNoErr(viper.BindPFlag("subdomains", flags.Lookup("subdomains")))
-
-	flags.String("assets", "", "path to the directory with the assets (use the packed assets by default)")
-	checkNoErr(viper.BindPFlag("assets", flags.Lookup("assets")))
-
-	flags.String("admin-host", "localhost", "administration server host")
-	checkNoErr(viper.BindPFlag("admin.host", flags.Lookup("admin-host")))
-
-	flags.Int("admin-port", 6060, "administration server port")
-	checkNoErr(viper.BindPFlag("admin.port", flags.Lookup("admin-port")))
```

Comment:
We should add `admin-host` and `admin-port` options the `cozy-stack instances` command